ReviewChecklist.md

```diff
@@ -0,0 +1,153 @@
+        Do this only for PRs created directly in the main repo.
+
+## Style and good practices
+- [ ] Does the PR follow our [coding style](CODING_STYLE.md)?
+
+### Reliability
+- [ ] **Use assertions liberally.** If you are certain your assumption will not be be broken, prove it with `solAssert()`.
+- [ ] **Validate inputs and handle errors**. Note that assertions are **not** validation.
+
+### Readability
+- [ ] **Choose good names.**
+    - [ ] Is the name straightforward to understand?
+        Do you feel the need to jump back to the definition and remind yourself what it was whenever you see it?
+    - [ ] Is the name unambiguous in context where it's used?
+    - [ ] Avoid abbreviations.
+- [ ] **Source files, classes and public functions should have docstrings.**
+- [ ] **Avoid code duplication.** But not fanatically. Minimal amounts of duplication are acceptable if it aids readability.
+- [ ] **Don't leave dead code behind.** You can still see old code in history.
+      But if you really have a good reason to do it, always leave a comment explaining what it is and why it's there.
+- [ ] **Mark hacks as such.** If you have to leave behind something ugly, make sure to include a
+    comment that explains why and in what circumstances it can be removed.
+    Preferably link to an issue you reported upstream.
+- [ ] **Avoid obvious comments.**
+- [ ] **Do include comments when the reader may need extra context to understand the code.**
+
+### Commits and PRs
+- [ ] **Avoid hiding functional changes inside refactors.**
+    E.g. when fixing a small bug, or changing user-visible behavior, put the change in a separate commit.
+    Do not mix it with another change that renames things or reformats the code around, making the fix itself hard to identify.
+- [ ] **Whenever possible, split off refactors or unrelated changes into separate PRs.**
+    Smaller PRs are easier and quicker to review.
+    Splitting off refactors helps focus on the main point of the PR.
+
+### Common pitfalls
+These are covered by the coding style but come up so often that it's worth singling them out here:
+- [ ] **Always initialize value types in the definition,** even if you're sure you will assign them later.
+- [ ] **Use "east const" style.** I.e.. `T const*`, not `const T *`.
+- [ ] **Indentation**. See our [`.editorconfig`](.editorconfig)).
+    - [ ] Tabs for C++. Use tabs for indentation only any whitespace later on the line must be spaces.
+    - [ ] 4 spaces for most other file types.
+- [ ] **Use `auto` sparingly.** Only when the actual type is very long and complicated or when it's
+    already used elsewhere in the same expression.
+- [ ] **Indent braces and parentheses in a way that makes nesting clear.**
+- [ ] **Use `using namespace` only in `.cpp` files.** Use it for `std` and our own modules.
+    Avoid unnecessary `std::` prefix in `.cpp` files (except for very common names like `std::move`).
```

Comment:
Eh, using just `move` will cause `check_style` to fail anyway.

Reply:
True. Do you suggest removing this mention?

I included it because it's still an important exception to the rule. Actually, did we even add the rule to `CODING_STYLE`?

Reply:
>True. Do you suggest removing this mention?

Not necessarily; I think adding a mention that using `move` without the `std::` qualifier will fail the check style job.

>I included it because it's still an important exception to the rule. Actually, did we even add the rule to CODING_STYLE?

I believe I did not  :smile:

Reply:
> I think adding a mention that using move without the std:: qualifier will fail the check style job.

But the submitter will easily discover that from the failing style check :) On the other hand the fact that `std::move` is an exception is not obvious.

Reply:
Yeah, I'll add it to codestyle docs.